pkg/ipmi/ipmi.go

```diff
@@ -173,3 +176,62 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (i *IPMI) getDeviceID() ([]byte, error) {
+	req := &req{}
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.cmd = _BMC_GET_DEVICE_ID
+
+	recv, err := i.sendrecv(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return recv, nil
+}
+
+func (i *IPMI) setGlobalEnables(enables byte) error {
+	req := &req{}
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.cmd = _BMC_SET_GLOBAL_ENABLES
+	req.msg.data = unsafe.Pointer(&enables)
+	req.msg.dataLen = 1
+
+	_, err := i.sendrecv(req)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *IPMI) getGlobalEnables() ([]byte, error) {
+	req := &req{}
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.cmd = _BMC_GET_GLOBAL_ENABLES
+
+	recv, err := i.sendrecv(req)
+	if err != nil {
+		return nil, err
+	}
+
+	return recv, nil
+}
+
+func (i *IPMI) EnableSEL() error {
+	// Check if SEL device is supported or not
+	if data, err := i.getDeviceID(); err != nil || (data[6]&0x04) == 0 {
```

Comment:
what's `data[6]`? why shouldn't it be 0x04? 

can you define some constants for it that name these things?